src/Common/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -35,7 +40,7 @@ internal static SafeFileHandle Open(string path, Interop.Sys.OpenFlags flags, in
             bool enoentDueToDirectory = (flags & Interop.Sys.OpenFlags.O_CREAT) != 0;
 
             // Open the file. 
-            int fd;
+            IntPtr fd;
```

Comment:
The part that hung me up was the `while (CheckIO(...))` loop.  If the open call gets interrupted a lot, won't we be creating and throwing away a lot of finalizable SafeHandle objects?  Is that a concern?